### PR TITLE
[Bigflix] Add new extractor for bigflix.com

### PR DESCRIPTION
**Pull request by vickyg3**
Add an IE to support bigflix.com. It uses some sort of silverlight
plugin whose video url is being populated using base64 encoded
flashvars. So it is quite straightforward to extract.

### CODE REVIEW

File: docs/supportedsites.md [hunk 1]
```diff
@@ -65,6 +65,7 @@
  - **Beeg**
  - **BehindKink**
  - **Bet**
+ - **Bigflix**
```
Comment (dstftw) [inline docs/supportedsites.md]: This file is autogenerated.

File: youtube_dl/extractor/bigflix.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from base64 import b64decode
+from urllib import unquote
```
Comment (dstftw) [inline youtube_dl/extractor/bigflix.py]: Use `compat_urllib_parse_unquote` instead.

File: youtube_dl/extractor/bigflix.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+                'title': 'Madarasapatinam',
+                'description': 'md5:63b9b8ed79189c6f0418c26d9a3452ca',
+            }
+        },
+    ]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        request = sanitized_Request(url)
```
Comment (dstftw) [inline youtube_dl/extractor/bigflix.py]: No need to use `sanitized_Request` request here, pass url directly to download method.

File: youtube_dl/extractor/bigflix.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        webpage = self._download_webpage(request, video_id)
+
+        title = self._html_search_regex(
+            r'<div[^>]+class=["\']pagetitle["\'][^>]*>(.+?)</div>',
+            webpage, 'title')
+
+        video_url = b64decode(unquote(self._search_regex(
```
Comment (dstftw) [inline youtube_dl/extractor/bigflix.py]: 1. `video_url` must be unicode string.
2. On python 3.2 `b64decode` accepts only bytes.

File: docs/supportedsites.md (hunk 1 above)
Comment (vickyg3) [inline docs/supportedsites.md]: Removed this from the PR. Would it be okay if i added this to .gitignore then?

File: youtube_dl/extractor/bigflix.py (hunk 2 above)
Comment (vickyg3) [inline youtube_dl/extractor/bigflix.py]: Done.

File: youtube_dl/extractor/bigflix.py (hunk 3 above)
Comment (vickyg3) [inline youtube_dl/extractor/bigflix.py]: Done.

File: youtube_dl/extractor/bigflix.py (hunk 4 above)
Comment (vickyg3) [inline youtube_dl/extractor/bigflix.py]: Fixed 2 by encoding the input to b64decode as ASCII.
Fixed 1 by encoding the output of b64decode as utf-8.

**Comment by vickyg3** [comment]
Thanks for the quick review. Have addressed your comments. Please take another look. :)

File: docs/supportedsites.md (hunk 1 above)
Comment (dstftw) [inline docs/supportedsites.md]: No, it's committed on every new release.

Comment (vickyg3) [inline docs/supportedsites.md]: alright.

File: youtube_dl/extractor/bigflix.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        video_id = self._match_id(url)
+
+        webpage = self._download_webpage(url, video_id)
+
+        title = self._html_search_regex(
+            r'<div[^>]+class=["\']pagetitle["\'][^>]*>(.+?)</div>',
+            webpage, 'title')
+
+        video_url = b64decode(compat_urllib_parse_unquote(self._search_regex(
+            r'file=([^&]+)', webpage, 'video url')).encode('ASCII')).encode('utf-8')
```
Comment (dstftw) [inline youtube_dl/extractor/bigflix.py]: It should be `decode('utf-8')` not encode. Also use lowercase for encoding name for consistency.

File: youtube_dl/extractor/bigflix.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+            'md5': 'ec76aa9b1129e2e5b301a474e54fab74',
+            'info_dict': {
+                'id': '16537',
+                'ext': 'mp4',
+                'title': 'Singham Returns',
+                'description': 'md5:3d2ba5815f14911d5cc6a501ae0cf65d',
+            }
+        },
+        {
+            'url': 'http://www.bigflix.com/Tamil-movies/Drama-movies/Madarasapatinam/16070',
```
Comment (dstftw) [inline youtube_dl/extractor/bigflix.py]: If both tests test the same functionality leave only one. Otherwise put short comment for additional test.

**Comment by dstftw** [comment]
Could you also squash commits after you fix everything?

File: youtube_dl/extractor/bigflix.py (hunk 5 above)
Comment (vickyg3) [inline youtube_dl/extractor/bigflix.py]: done.

File: youtube_dl/extractor/bigflix.py (hunk 6 above)
Comment (vickyg3) [inline youtube_dl/extractor/bigflix.py]: done.

**Comment by vickyg3** [comment]
Have squashed everything into a single commit.

**Comment by dstftw** [comment]
Thanks.

**Pull request merged**